samcli/local/init/arbitrary_project.py

```diff
@@ -89,6 +91,6 @@ def _download_and_copy(download_fn, output_dir):
 
     with osutils.mkdir_temp() as tempdir:
         downloaded_dir = download_fn(clone_to_dir=tempdir)
-        osutils.copytree(downloaded_dir, output_dir)
+        osutils.copytree(downloaded_dir, output_dir, ignore=shutil.ignore_patterns("*.git"))
```

Comment:
Are we sure this works as we expect? This looks like the issue the Ruby builder had: https://github.com/awslabs/aws-lambda-builders/pull/130

Reading docs, this should work but maybe I just misunderstand the ruby builder fix.

Reply:
yeah we use this the same way lambda builders does. It supports glob patterns so we don't copy .git from anywhere, including submodules.